src/components/Filter/common/utils.ts

```diff
@@ -74,7 +74,7 @@ export function getSortMatchesFn(
       let match = matchString(value.label || "");
       if (match) {
         matches.push({ labelRanges: match.ranges, score: match.score, value });
-      } else {
+      } else if (typeof value.key === "string") {
```

Comment:
Do we care whether non-string underlying values get searched in filter search? I've made it so they don't (but the label, which is a string (normally?) derived from the same value, is)